apis/appmesh/v1beta2/mesh_types.go

```diff
@@ -81,6 +81,14 @@ type MeshSpec struct {
 	// Required if the account ID is not your own.
 	// +optional
 	MeshOwner *string `json:"meshOwner,omitempty"`
+	// +optional
+	MeshServiceDiscovery *MeshServiceDiscovery `json:"meshServiceDiscovery,omitempty"`
```

Comment:
The field names should align with appmesh sdk wherever possible.

Reply:
AppMesh SDK is using `MeshServiceDiscovery` in the field.